### PR TITLE
#genzsprthree-15 OOP-Based Contact Book Management System Redesign

### PR DESCRIPTION
**Pull request by jnulia**

### CODE REVIEW

**Comment by jnulia** [comment]
[OOP-Based Contact Book Management System Redesign](https://trello.com/c/MGNY5O8x/15-oop-based-contact-book-management-system-redesign)

**Comment by jnulia** [comment]
[OOP-Based Contact Book Management System Redesign](https://trello.com/c/MGNY5O8x/15-oop-based-contact-book-management-system-redesign)

**Comment by jnulia** [comment]
[OOP-Based Contact Book Management System Redesign](https://trello.com/c/MGNY5O8x/15-oop-based-contact-book-management-system-redesign)

**Comment by jnulia** [comment]
[OOP-Based Contact Book Management System Redesign](https://trello.com/c/MGNY5O8x/15-oop-based-contact-book-management-system-redesign)

**Comment by jnulia** [comment]
[OOP-Based Contact Book Management System Redesign](https://trello.com/c/MGNY5O8x/15-oop-based-contact-book-management-system-redesign)

**Review by Sughiya-AlSaid — DISMISSED** [review]

File: examples/lujain_work_area/Contact_Book_OOP/app_user.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# app_user.py
+
+from contact_book import ContactBook
```
Comment (CodelineAtyab) [inline examples/lujain_work_area/Contact_Book_OOP/app_user.py:3]: I am unable to find ContactBook class in contact_book.py

File: examples/lujain_work_area/Contact_Book_OOP/contact_book.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        for contact in self.contacts:
+            if any(contact.__dict__[key] == identifier for key in contact.__dict__):
+                for key, value in new_contact.items():
+                    setattr(contact, key, value)
+
+    def search_contact(self, identifier):
+        """Search for a contact in the contact book."""
+        return [contact for contact in self.contacts if any(contact.__dict__[key] == identifier for key in contact.__dict__)]
+
+from json_contact_book import JSONContactBook  # Importing JSONContactBook class
```
Comment (CodelineAtyab) [inline examples/lujain_work_area/Contact_Book_OOP/contact_book.py:36]: Please specify, Is there any specific reason for importing these classes at the end ?

File: examples/lujain_work_area/Contact_Book_OOP/app_user.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        """Update a contact in the user's ContactBook."""
+        self.contact_book.update_contact(identifier, new_contact)
+
+    def search_contact(self, identifier):
+        """Search for a contact in the user's ContactBook."""
+        return self.contact_book.search_contact(identifier)
+
+    def save_contact_book(self, filename, format):
+        """Save the user's ContactBook to a file."""
+        if format == 'csv':
```
Comment (CodelineAtyab) [inline examples/lujain_work_area/Contact_Book_OOP/app_user.py:29]: We can avoid this if else, if we use anstract methods and the implementation that is going to be used will be decided dynamically

File: examples/lujain_work_area/Contact_Book_OOP/app_user.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+            self.contact_book.save_to_csv(filename)
+        elif format == 'json':
+            self.contact_book.save_to_json(filename)
+
+    def load_contact_book(self, filename, format):
+        """Load the user's ContactBook from a file."""
+        if format == 'csv':
```
Comment (CodelineAtyab) [inline examples/lujain_work_area/Contact_Book_OOP/app_user.py:36]: We can just call load and not worry about, if its csv or json.

We have to point to the right Object in __init__, we can just put conditions there and then we would not need to use conditions anywhere else if we want to load or save

**Review by HaithamAlMaamari — DISMISSED** [review]

**Review by Sughiya-AlSaid — APPROVED** [review]

**Review by sarah-alshukaily — APPROVED** [review]

**Review by aljab017 — APPROVED** [review]

**Review by Rudainasaleh — APPROVED** [review]

**Pull request merged**